test(chat): add a test that install a sticker pack and sends a sticker

jrainville opened the pull request:
Fixes #6962

Branched on top of the IPFS gateway fix

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 0696f42a | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-6989/1/) | 2022-08-11 19:57:38 | ~8 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220811-194846-0696f4-pr6989.dmg) |

noeliaSD commented on test/ui-test/testSuites/suite_status/tst_ChatFlow/test.feature:
```diff
@@ -54,3 +54,9 @@ Feature: Status Desktop Chat
+    Scenario: User can send a sticker after installing a free pack
+         When user joins chat room automation-test
+         Then the user can install the sticker pack with id 15
+         Then the user can send the sticker at index 2
```
Just a minor comment here.. Reading it without context, I'm not sure what does it mean sending the sticker at `index 2`. It seems something about code not something that the user should be able to validate. Could we reformulate it to be more "user jargon"?

noeliaSD reviewed (commented):
When I run the test it happens the following (in this record `mailserver` is not loading the chat but the same behavior happend when the chat was loaded):

https://user-images.githubusercontent.com/97019400/184911195-0c050554-ba43-4166-bb9a-9c210e5726ee.mov

jrainville commented:
@noeliaSD I rebased and tested again. It worked fine for me. Make sure that status-go is rebuilt. The old gateway for stickers was broken. If it still doesn't work for you, can you try to just open the app and see if it ends up loading, but just very slow?

noeliaSD commented:
> @noeliaSD I rebased and tested again. It worked fine for me. Make sure that status-go is rebuilt. The old gateway for stickers was broken. If it still doesn't work for you, can you try to just open the app and see if it ends up loading, but just very slow?

I've tested again and same happend. Also tested manually, one time it loaded quick but the majority of times it keeps loading like in case of the auto-test.

jrainville commented:
@noeliaSD if you don't mind, I'd like to pair with you to see if I can understand what's going wrong and if I can fix it.

noeliaSD commented:
> @noeliaSD if you don't mind, I'd like to pair with you to see if I can understand what's going wrong and if I can fix it.

Sure thing!

borismelnik commented on ci/Jenkinsfile.uitests:
```diff
@@ -27,6 +27,14 @@ pipeline {
+      categories: ['status-desktop-e2e-tests'],
+      maxConcurrentPerNode: 1,
+      maxConcurrentTotal: 1
+    )
```
Are you sure that it's a good idea for this PR? For example, I'm can't give a good review for CI and it can be merged without `good review`

borismelnik commented on ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml:
```diff
@@ -36,79 +37,61 @@ Item {
-        visible: contactListRoot.title !== ""
-        text: contactListRoot.title
+        visible: contactsList.count > 0 && root.title !== ""
+        text: root.title
```
`property alias title:  titleStyledText.text` :) And pls, if you have time fix `id: title` (I see a conflict with `root.title`)

borismelnik commented on ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml:
```diff
@@ -0,0 +1,96 @@
+Item {
+    id: root
+
+    property alias displayName: displayNameInput
```
Could you pls change it to more suitable name, cos `displayName` means `text` not component

borismelnik reviewed (changes requested):
Super :) just few questions

jrainville commented on test/ui-test/src/drivers/SquishDriver.py:
```diff
@@ -116,6 +116,10 @@ def click_obj_by_name(objName: str):
     obj = squish.waitForObject(getattr(names, objName))
     squish.mouseClick(obj, squish.Qt.LeftButton)
 
+def click_obj_by_attr(attr: str):
```
It's the attribute (json object) that comes from the names map. I use `getattr` to get it (https://www.programiz.com/python-programming/methods/built-in/getattr)
and then I replace the placeholder by the emoji name.
See StatusChatScreen line 266.

To be honest, I didn't know what name to give it, because it's not a name, and it's not an actual Qt object either.

jrainville commented:
@borismelnik sorry about the messed up rebase. You probably reviewed files that had nothing to do with my PR.
You can check again, the rebase is fixed.

jrainville commented:
Can I get a review here please

noeliaSD reviewed (commented):
For me now works :partying_face: BUT not always. 

During testing I can only see the loading icons as it is faster finishing the test than actually loading the stickers. Could this cause the first issue of the video? Maybe we need to smooth the test in some points.

https://user-images.githubusercontent.com/97019400/186361566-32ad2731-948f-48bf-a1c8-06bebf1dd273.mov

jrainville commented:
@noeliaSD I updated it. I think the issue was that you have a different order of sticker packs as me, so I changed it to selecting the pack by index and using `scroll_list_view_at_index`. 
Even if the sticker keeps loading (the image), it still works.

noeliaSD commented on .gitignore:
```diff
@@ -41,7 +41,7 @@ nim_status_client.log
 **/fixtures/*/ipfs
 
 test/ui-test/testSuites/suite_status/config.xml
```
Should it need the same here?
```suggestion
test/ui-test/testSuites/*/config.xml
```

jrainville replied:
Yeah, but it's not necessary to amend this commit further. It can be changed in one of my other PRs

noeliaSD reviewed (approved):
LGTM!
